signer/src/stacks/api.rs

```diff
@@ -634,76 +626,116 @@ impl StacksClient {
+
+    Ok(blocks)
+}
+
+/// A deserializer for Clarity's [`Value`] type that deserializes a hex-encoded
+/// string which was serialized using Clarity's consensus serialization format.
+fn clarity_value_deserializer<'de, D>(deserializer: D) -> Result<Value, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Value::try_deserialize_hex_untyped(&String::deserialize(deserializer)?)
+        .map_err(serde::de::Error::custom)
 }
 
 impl StacksInteract for StacksClient {
     async fn get_current_signer_set(
-        &mut self,
+        &self,
         contract_principal: &StacksAddress,
     ) -> Result<Vec<PublicKey>, Error> {
         // Make a request to the sbtc-registry contract to get the current
         // signer set.
-        let result = StacksClient::get_data_var(
-            self,
-            contract_principal,
-            &ContractName::from("sbtc-registry"),
-            &ClarityName::from("current-signer-set"),
-        )
-        .await;
-
-        let checked_result = check_result(self, result)?;
+        let result = self
+            .get_data_var(
+                contract_principal,
+                &ContractName::from("sbtc-registry"),
+                &ClarityName::from("current-signer-set"),
+            )
+            .await?;
 
-        match checked_result {
+        // Check the result and return the signer set. We're expecting a
+        // list of buffers, where each buffer is a public key.
+        match result {
             Value::Sequence(SequenceData::List(ListData { data, .. })) => {
-                let mut keys = Vec::new();
-                for item in data {
-                    if let Value::Sequence(SequenceData::Buffer(BuffData { data })) = item {
-                        keys.push(PublicKey::from_slice(&data)?);
-                    } else {
-                        return Err(Error::InvalidStacksResponse(
+                // Iterate through each record in the list and verify that it's a buffer.
+                // If it is a buffer, then convert it to a public key.
+                // Otherwise, abort retries and return an error.
+                data.into_iter()
+                    .map(|item| match item {
+                        // If the item is a buffer, then convert it to a public key.
+                        Value::Sequence(SequenceData::Buffer(BuffData { data })) => {
+                            PublicKey::from_slice(&data)
+                        }
+                        // Otherwise, return an error.
+                        _ => Err(Error::InvalidStacksResponse(
                             "expected a buffer but got something else",
-                        ));
-                    }
-                }
-                Ok(keys)
+                        )),
+                    })
+                    .collect()
             }
+            // We expected the top-level value to be a list of buffers,
+            // but we got something else. Abort retrying and return an error.
```

Comment:
Probably just delete the last sentence?

Reply:
Addressed in [5f48dd2](https://github.com/stacks-network/sbtc/pull/572/commits/5f48dd257ab134735de76327fa520789f57ed27d)